### PR TITLE
[PM-17242] While on autofill search on all item types.

### PR DESCRIPTION
**Pull request by andrebispo5**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-17242
## 📔 Objective

Allow search terms to be applied on all item types during autofill.

## 📸 Screenshots

https://github.com/user-attachments/assets/db81e08f-7e4a-4499-b44e-7a5865c4e187

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt [hunk 1]
```diff
@@ -1049,9 +1049,15 @@ class VaultItemListingViewModel @Inject constructor(
     }
 
     private fun handleSearchIconClick() {
+        val searchType = if (state.autofillSelectionData != null) {
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt:1052]: Could we add a test for this case in `VaultItemListingViewModelTest` for when the autofillSelectionData is not null?

**Review by dseverns-livefront — APPROVED** [review]

File: app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModelTest.kt [hunk 2]
```diff
@@ -339,6 +339,24 @@ class VaultItemListingViewModelTest : BaseViewModelTest() {
         }
     }
 
+    @Test
+    fun `SearchIconClick should emit NavigateToVaultSearchScreen with all search type`() = runTest {
```
Comment (dseverns-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModelTest.kt:343]: 👍
